Archive/Imo/Imo1982Q1.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2024 Alex Brodbelt. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Brodbelt
+-/
+
+import Mathlib.Tactic
+/-!
+# IMO 1982 Q1
+
+The function $f(n)$ is defined for all postive integers $n$ and takes on
+non-negative integer values. Also, for all m, n
+
+$f (m + n) - f(m) - f(n) = 0 \text{ or } 1$
+$f(2) = 0, f(3) > 0, and f(9999) = 3333.$
+
+Determine f(1982).
+
+The solution is based on the one at the
+[Art of Problem Solving](https://artofproblemsolving.com/wiki/index.php/1982_IMO_Problems/Problem_1)
+website.
+-/
+
+namespace Imo1982Q1
+
+structure IsGood (f : ℕ → ℕ) : Prop where
+  /-- Satisfies the functional relation-/
+  rel: ∀ m n : ℕ, f (m + n) = f m + f n ∨ f (m + n) = f m + f n + 1
+  f₂ : f 2 = 0
+  hf₃ : f 3 > 0
+  f_9999 : f 9999 = 3333
+
+namespace IsGood
+
+variable {f : ℕ → ℕ} (hf : IsGood f)
+include hf
+
+lemma f₀ : f 0 = 0 := by
```

Comment:
This lemma doesn't make sense. In the original problem, `f 0` isn't even defined, so making use of its value is something we shouldn't be doing.